src/integrationtests/java/com/aws/greengrass/integrationtests/SecretManagerServiceIntegTest.java

```diff
@@ -0,0 +1,196 @@
+    private Kernel kernel;
+    private final SecurityService mockSecurityService = spy(new SecurityService(mock(DeviceConfiguration.class)));
+
+    @TempDir
+    Path rootDir;
+
+    @Mock
+    AWSSecretClient secretClient;
+
+    void startKernelWithConfig(String configFile, State expectedState) throws Exception {
+        URI privateKey = getClass().getResource("privateKey.pem").toURI();
+        URI certUri = getClass().getResource("cert.pem").toURI();
+        lenient().doReturn(privateKey).when(mockSecurityService).getDeviceIdentityPrivateKeyURI();
+        lenient().doReturn(certUri).when(mockSecurityService).getDeviceIdentityCertificateURI();
+        lenient().doReturn(EncryptionUtils.loadPrivateKeyPair(Paths.get(privateKey))).when(mockSecurityService).getKeyPair(privateKey, certUri);
+        mockSecretResponse();
+        kernel = new Kernel();
+        kernel.parseArgs("-r", rootDir.toAbsolutePath().toString(), "-i",
+                getClass().getResource(configFile).toString());
+
+        CountDownLatch secretManagerRunning = new CountDownLatch(1);
+        kernel.getContext().addGlobalStateChangeListener((GreengrassService service, State was, State newState) -> {
+            if (service.getName().equals(SecretManagerService.SECRET_MANAGER_SERVICE_NAME) && service.getState()
+                    .equals(expectedState)) {
+                secretManagerRunning.countDown();
+            }
+        });
+        kernel.getContext().put(AWSSecretClient.class, secretClient);
+        kernel.getContext().put(SecurityService.class, mockSecurityService);
+        kernel.launch();
+
+        assertTrue(secretManagerRunning.await(10, TimeUnit.SECONDS));
+    }
+
+    private void mockSecretResponse() throws SecretManagerException, IOException {
+        String secretArn = "arn:aws:secretsmanager:us-east-1:999936977227:secret:randomSecret-74lYJh";
+        String secretName = "randomSecret";
+        lenient().doReturn(software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse.builder()
+                .name(secretName).arn(secretArn).secretString("secretValue").versionId(VERSION_ID)
+                .versionStages(CURRENT_LABEL)
+                .createdDate(Instant.now().minusSeconds(1000000)).build())
+                .when(secretClient).getSecret(GetSecretValueRequest.builder().secretId(secretArn).versionStage(CURRENT_LABEL).build());
+
+        lenient().doReturn(software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse.builder()
+                .name(secretName).arn(secretArn).secretString("secretValue2").versionId("id2")
+                .versionStages("new").createdDate(Instant.now().minusSeconds(1000000)).build())
+                .when(secretClient).getSecret(GetSecretValueRequest.builder().secretId(secretArn).versionStage("new").build());
+    }
+
+    @AfterEach
+    void cleanup() {
+        kernel.shutdown();
+    }
+
+    @BeforeEach
+    void setup(ExtensionContext context) {
```

Comment:
extend from BaseITCase to get a lot of this for free

Reply:
Updated but I still have to do this setup.